src/background/stores/assessment-card-selection-store.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { AssessmentCardSelectionActions } from 'background/actions/assessment-card-selection-actions';
+import { IndexedDBDataKeys } from 'background/IndexedDBDataKeys';
+import { PersistentStore } from 'common/flux/persistent-store';
+import { IndexedDBAPI } from 'common/indexedDB/indexedDB';
+import { Logger } from 'common/logging/logger';
+import { AssessmentCardSelectionStoreData } from 'common/types/store-data/assessment-card-selection-store-data';
+import { forOwn, isEmpty } from 'lodash';
+import { StoreNames } from '../../common/stores/store-names';
+import {
+    CardSelectionStoreData,
+    RuleExpandCollapseData,
+} from '../../common/types/store-data/card-selection-store-data';
+import {
+    AssessmentCardSelectionPayload,
+    AssessmentCardToggleVisualHelperPayload,
+    AssessmentExpandCollapsePayload,
+    AssessmentNavigateToNewCardsViewPayload,
+    AssessmentResetFocusedIdentifierPayload,
+    RuleExpandCollapsePayload,
+} from '../actions/action-payloads';
+
+export class AssessmentCardSelectionStore extends PersistentStore<AssessmentCardSelectionStoreData> {
+    constructor(
+        private readonly assessmentCardSelectionActions: AssessmentCardSelectionActions,
+        persistedState: AssessmentCardSelectionStoreData,
+        idbInstance: IndexedDBAPI,
+        logger: Logger,
+        tabId: number,
+        persistStoreData: boolean,
+    ) {
+        super(
+            StoreNames.AssessmentCardSelectionStore,
+            persistedState,
+            idbInstance,
+            IndexedDBDataKeys.assessmentCardSelectionStore(tabId),
+            logger,
+            persistStoreData,
+        );
+    }
+
+    protected addActionListeners(): void {
+        this.assessmentCardSelectionActions.toggleRuleExpandCollapse.addListener(
+            this.toggleRuleExpandCollapse,
+        );
+        this.assessmentCardSelectionActions.toggleCardSelection.addListener(
+            this.toggleCardSelection,
+        );
+        this.assessmentCardSelectionActions.collapseAllRules.addListener(this.collapseAllRules);
+        this.assessmentCardSelectionActions.expandAllRules.addListener(this.expandAllRules);
+        this.assessmentCardSelectionActions.toggleVisualHelper.addListener(this.toggleVisualHelper);
+        this.assessmentCardSelectionActions.getCurrentState.addListener(this.onGetCurrentState);
+        this.assessmentCardSelectionActions.resetFocusedIdentifier.addListener(
+            this.onResetFocusedIdentifier,
+        );
+        this.assessmentCardSelectionActions.navigateToNewCardsView.addListener(
+            this.onNavigateToNewCardsView,
+        );
+    }
+
+    public getDefaultState(): AssessmentCardSelectionStoreData {
+        const defaultValue: AssessmentCardSelectionStoreData = {
+            'automated-checks': {
```

Comment:
I haven't given much thought to what the default state should be, but I had assumed that it would be `{}` and we would add testKeys along the way. What is the rationale behind including the `automated-checks` value from the beginning?

Reply:
I just wanted to have something there for now but I can take it out. I will be adding the store initialization logic in another PR.